src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = true;
+    Animal.alive.push(this);
+  }
+
+  bite(someone) {
+    if (someone.hidden === false) {
```

Comment:
```suggestion
    if (!someone.hidden) {
```
Again, let's read the description:

> Carnivore has a `bite` method, which takes a `herbivore` object and decreases the object's health by 50

As you may see, `bite` method should only bite herbivores. Now it can bite anyone (herbivores and carnivores both). Let's add some new check and fix this